src/lib/dnssd/ActiveResolveAttempts.cpp

```diff
@@ -53,7 +53,30 @@ void ActiveResolveAttempts::Complete(const PeerId & peerId)
     ChipLogProgress(Discovery, "Discovered node without a pending query");
 }
 
-void ActiveResolveAttempts::MarkPending(const PeerId & peerId)
+void ActiveResolveAttempts::Complete(const chip::Dnssd::DiscoveredNodeData & data)
+{
+    for (auto & item : mRetryQueue)
+    {
+        if (item.attempt.Matches(data))
+        {
+            item.attempt.Clear();
+            return;
+        }
+    }
+}
+
+void ActiveResolveAttempts::MarkPending(const chip::PeerId & peerId)
+{
+    ScheduledAttempt attempt(peerId, true);
```

Comment:
reading through the code, the `true` here and below is not obvious and makes me wonder "what does an attempt being true mean... or a peerId being true". Please use some enum constants or at least add a `/**/` comment to say what the value means.